src/vorta/views/partials/file_dialog.py

```diff
@@ -0,0 +1,145 @@
+        self.sortproxy.searchStringError.connect(self.searchStringError)
+
+        self.buttonBox.accepted.connect(self.accept)
+        self.buttonBox.rejected.connect(self.reject)
+
+        # Add a cross icon inside the search field to clear the search string
+        self.searchWidget.setClearButtonEnabled(True)
+        # self.searchLineEdit.textChanged.connect(self.searchLineEditChanged)
+
+        self.set_icons()
+
+        # Connect to palette change
+        QApplication.instance().paletteChanged.connect(lambda p: self.set_icons())
+
+    @abstractmethod
+    def get_sort_proxy_model(self):
+        pass
+
+    @abstractmethod
+    def get_diff_result_display_mode(self):
+        pass
+
+    @abstractmethod
+    def set_archive_names(self):
+        pass
+
+    def copy_item(self, index: QModelIndex = None):
+        """
+        Copy a diff item path to the clipboard.
+
+        Copies the first selected item if no index is specified.
+        """
+        if index is None or (not index.isValid()):
+            indexes = self.treeView.selectionModel().selectedRows()
+
+            if not indexes:
+                return
+
+            index = indexes[0]
+
+        index = self.sortproxy.mapToSource(index)
+        item = index.internalPointer()
+        path = PurePath('/', *item.path)
+
+        data = QMimeData()
+        data.setUrls([QUrl(path.as_uri())])
+        data.setText(str(path))
+
+        QApplication.clipboard().setMimeData(data)
+
+    def treeview_context_menu(self, pos: QPoint):
+        """Display a context menu for `treeView`."""
+        index = self.treeView.indexAt(pos)
+        if not index.isValid():
+            # popup only for items
+            return
+
+        menu = QMenu(self.treeView)
+
+        menu.addAction(get_colored_icon('copy'), self.tr("Copy"), lambda: self.copy_item(index))
+
+        if self.model.getMode() != self.model.DisplayMode.FLAT:
+            menu.addSeparator()
+            menu.addAction(
+                get_colored_icon('angle-down-solid'),
+                self.tr("Expand recursively"),
+                lambda: self.treeView.expandRecursively(index),
+            )
+
+        menu.popup(self.treeView.viewport().mapToGlobal(pos))
+
+    def slot_sorted(self, column, order):
+        """React to the tree view being sorted."""
+        # reveal selection
+        selectedRows = self.treeView.selectionModel().selectedRows()
+        if selectedRows:
+            self.treeView.scrollTo(selectedRows[0])
+
+    def keyPressEvent(self, event):
+        if event.key() in [Qt.Key.Key_Return, Qt.Key.Key_Enter] and self.searchWidget.hasFocus():
+            self.submitSearchPattern()
+        else:
+            super().keyPressEvent(event)
+
+    def submitSearchPattern(self):
+        self.sortproxy.setSearchString(self.searchWidget.text())
+
+    def searchStringError(self, error: bool):
+        self.searchWidget.setStyleSheet("border: 2px solid red;" if error else "")
```

Comment:
This leads to the same issue as in a previous PR: The tooltip also gets the border.